payload/message.go

```diff
@@ -37,49 +39,68 @@ type (
 
 		hash *util.Uint256
 	}
+
+	// payloadAux is an auxiliary structure for Payload encoding.
+	payloadAux struct {
+		Version        uint32
+		ValidatorIndex uint16
+		PrevHash       util.Uint256
+		Height         uint32
+
+		Data []byte
+	}
 )
 
 var _ ConsensusPayload = (*Payload)(nil)
 
-// EncodeBinary implements io.Serializable interface.
-func (p Payload) EncodeBinary(w *io.BinWriter) {
-	ww := io.NewBufBinWriter()
-	p.message.EncodeBinary(ww.BinWriter)
-	data := ww.Bytes()
-
-	w.WriteU32LE(p.version)
-	w.WriteBytes(p.prevHash[:])
-	w.WriteU32LE(p.height)
-	w.WriteU16LE(p.validatorIndex)
-	w.WriteVarBytes(data)
+// EncodeBinary implements Serializable interface.
+func (p Payload) EncodeBinary(w *gob.Encoder) error {
```

Comment:
And this structure as well, who needs it exported?

Reply:
But how about some real intermediate implementation, like we had in Geth?